base/range.jl

```diff
@@ -47,23 +47,48 @@ function _colon(start::T, step, stop::T) where T
 end
 
 """
-    range(start[, stop]; length, stop, step=1)
-
-Given a starting value, construct a range either by length or from `start` to `stop`,
-optionally with a given step (defaults to 1, a [`UnitRange`](@ref)).
-One of `length` or `stop` is required.  If `length`, `stop`, and `step` are all specified, they must agree.
+    range(start; stop)
+    range(start; length)
+    range(start, stop; length)
+    range(start, stop; step)
+
+    # The following cause an ArgumentError to be thrown.
+    range(start, stop) # Must specify either length or step
+    range(start; step) # Cannot specify step alone
+    range(start, stop; length, step) # Cannot specify all of stop, length, and step
+    range(start; stop, length, step) # Cannot specify all of stop, length, and step
+
```

Comment:
I believe we should always put the one-or-two-line description right after the function signature.

This can be moved backward in the Examples section.